### PR TITLE
fix(context): fall back to default memory bootstrap

### PR DESCRIPTION
**Pull request by yu-xin-c**
## Summary

- load project-local `AGENTS.md` as before while falling back to the default workspace for missing `SOUL.md` and `USER.md`
- keep project-local `SOUL.md` / `USER.md` files preferred when they exist
- add regression coverage for bare project workspaces and project-local overrides

This addresses the fallback/no-persona facet described in #4374 without changing Dream's shared memory/history write root.

## Tests

- `uv run pytest tests/agent/test_context_builder.py -q`
- `uv run ruff check nanobot/agent/context.py tests/agent/test_context_builder.py`
- `git diff --check`
- `uv run pytest tests/agent/test_context_builder.py tests/agent/test_workspace_scope.py tests/channels/test_websocket_channel.py -q`
- `uv run pytest tests/agent/test_context_builder.py tests/agent/test_dream.py -q`

### CODE REVIEW

File: nanobot/agent/context.py [hunk 1]
```diff
@@ -164,18 +165,32 @@ def _to_blocks(value: Any) -> list[dict[str, Any]]:
...
 
         return "\n\n".join(parts) if parts else ""
 
+    def _resolve_bootstrap_file(self, root: Path, filename: str) -> Path | None:
+        file_path = root / filename
+        if file_path.exists():
+            return file_path
+        if filename not in self._DEFAULT_FALLBACK_BOOTSTRAP_FILES:
+            return None
+        if root.expanduser().resolve(strict=False) == self.workspace.expanduser().resolve(strict=False):
```
Comment (chengyongru) [inline nanobot/agent/context.py:187]: This self-loop guard prevents fallback when root equals the default workspace. Could you add a test for this path (calling `_load_bootstrap_files()` without a workspace argument when bootstrap files exist)? It's a defensive edge case that would benefit from explicit coverage.

**Review by chengyongru — COMMENTED** [review]
Automated PR review by nanobot. This is not a human maintainer review or approval.

## Summary

Addresses the fallback side of #4374: when a project-scoped workspace lacks `SOUL.md` or `USER.md`, the `ContextBuilder` falls back to reading them from the default workspace. `AGENTS.md` is excluded from the fallback (it is inherently project-specific).

## What is done well

- **Minimal change**: 20 additions, 5 deletions, entirely within `context.py`. Extracts file resolution into a dedicated `_resolve_bootstrap_file` method.
- **Correct filtering**: `_DEFAULT_FALLBACK_BOOTSTRAP_FILES` excludes `AGENTS.md` — project rules should never come from the default workspace.
- **Self-loop guard**: `root == self.workspace` check prevents fallback recursion when `_load_bootstrap_files` is called without explicit workspace.
- **Tests cover both scenarios**: bare project workspace (fallback from default) and project-local override (takes priority over default).

## Concerns

1. **Read/write asymmetry remains for project-local files**: If a user intentionally places `SOUL.md`/`USER.md` in a project workspace, reads come from the project path but Dream's `MemoryStore` writes to the default workspace. The PR acknowledges this limitation explicitly. This is acceptable as an incremental step — the bare workspace case is the most common scenario.

2. **No regression test for the self-loop guard**: The `root == self.workspace` branch in `_resolve_bootstrap_file` (line 187) isn't covered by a test. An explicit test calling `_load_bootstrap_files()` without a workspace argument when both default and project paths exist would be a useful safety net.

No blocking concerns.

**Comment by yu-xin-c** [comment]
Closing this to keep the PR queue focused. The original context issue (#4374) was closed after being reclassified, so this fallback behavior is better revisited only if maintainers want it as an explicit feature.
